crates/bevy_asset/src/assets.rs

```diff
@@ -590,6 +595,47 @@ impl<A: Asset> Assets<A> {
         }
     }
 
+    pub(crate) fn queue_untyped_asset_modified_events(
+        assets: Res<Self>,
+        mut untyped_asset_modified_events: EventWriter<UntypedAssetModifiedEvent>,
+    ) {
+        untyped_asset_modified_events.write_batch(assets.queued_events.iter().filter_map(|e| {
+            if let AssetEvent::Modified { id } = e {
+                Some(UntypedAssetModifiedEvent { id: id.untyped() })
+            } else {
+                None
+            }
+        }));
+    }
+
+    pub(crate) fn handle_asset_dependency_modified_events(
+        mut assets: ResMut<Self>,
+        mut untyped_asset_modified_events: EventReader<UntypedAssetModifiedEvent>,
+    ) {
+        if untyped_asset_modified_events.is_empty() {
+            return;
+        }
+        let untyped_asset_modified_events: HashSet<_> =
+            untyped_asset_modified_events.read().map(|e| e.id).collect();
+        let mut modified_events = Vec::new();
+        for (id, asset) in assets.iter() {
```

Comment:
Oh this literally iterates through every single asset and checking if it (directly or indirectly) depends on one of the modified assets. I think this is a dealbreaker for me - iterating through **every** asset seems like it'll be very slow.